handsontable/src/editors/autocompleteEditor/autocompleteEditor.js

```diff
@@ -110,15 +110,15 @@ export class AutocompleteEditor extends HandsontableEditor {
       renderer: (instance, TD, row, col, prop, value, cellProperties) => {
         textRenderer(instance, TD, row, col, prop, value, cellProperties);
 
-        const { filteringCaseSensitive, allowHtml } = this.cellProperties;
+        const { filteringCaseSensitive, allowHtml, locale } = this.cellProperties;
         const query = this.query;
         let cellValue = stringify(value);
         let indexOfMatch;
         let match;
 
         if (cellValue && !allowHtml) {
           indexOfMatch = filteringCaseSensitive === true ?
-            cellValue.indexOf(query) : cellValue.toLowerCase().indexOf(query.toLowerCase());
+            cellValue.indexOf(query) : cellValue.toLocaleLowerCase(locale).indexOf(query.toLocaleLowerCase(locale));
```

Comment:
Could you add a test case for that changes?

Reply:
I'd rather not add a test as it uses internally `sort` method with custom comparing function which isn't perfect (it doesn't compare locale-based data properly; it should be rewrote the same as default sort method, but it is out of scope). This change just corrects a little bit the code.

Reply:
Ok, so `this PR doesn't change a way how sorting works.` refers to that also. That's fine :+1: